test/modules/obsagent-client/src/main.rs

```diff
@@ -0,0 +1,204 @@
+use futures::stream::Stream;
+use futures::StreamExt;
+// use opentelemetry::global::shutdown_tracer_provider;
+use opentelemetry::sdk::metrics::{selectors, PushController};
+// use opentelemetry::trace::TraceError;
+use opentelemetry::global;
+use opentelemetry::metrics::{self, ObserverResult, ValueRecorder};
+use opentelemetry_otlp::ExporterConfig;
+use std::error::Error;
+use std::time::Duration;
+
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::{Arc, Mutex};
+
+use clap::{value_t, App, Arg};
+use rand::Rng;
```

Comment:
The style we follow for these deps is the order:
1. std
2. external crates
3. internal crates
4. internal `use crate` declarations

Here is an example:
```
use std::{
    cmp,
    collections::{HashMap, VecDeque},
    error::Error as StdError,
    fs::{self, OpenOptions},
    io::{Read, Write},
    path::PathBuf,
};

use async_trait::async_trait;
use bytes::Bytes;
use chrono::{DateTime, Utc};
use fail::fail_point;
use flate2::{read::GzDecoder, write::GzEncoder, Compression};
use serde::{ser::SerializeMap, Deserialize, Deserializer, Serialize, Serializer};
use tracing::{debug, info, info_span};

use mqtt3::proto::{Publication, Publish};

use crate::{
    proto::{PacketIdentifierDupQoS, QoS},
    subscription::Subscription,
    BrokerSnapshot, ClientInfo, SessionSnapshot,
};
```